lib/rules/disallow-curly-braces.js

```diff
@@ -0,0 +1,149 @@
+/**
+ * Disallows curly braces after statements.
+ *
+ * Type: `Array` or `Boolean`
+ *
+ * Values: Array of quoted keywords or `true` to disallow curly braces after the following keywords:
+ *
+ * #### Example
+ *
+ * ```js
+ * "disallowCurlyBraces": [
+ *     "if",
+ *     "else",
+ *     "for",
+ *     "while",
+ *     "do",
+ *     "try",
+ *     "catch",
+ *     "case",
+ *     "default"
+ * ]
+ * ```
+ *
+ * ##### Valid
+ *
+ * ```js
+ * if (x) {
+ *     x++;
+ * }
+ * ```
+ *
+ * ##### Invalid
+ *
+ * ```js
+ * if (x) x++;
+ * ```
+ */
+
+var assert = require('assert');
+var defaultKeywords = require('../utils').curlyBracedKeywords;
+
+module.exports = function() {};
+
+module.exports.prototype = {
+
+    configure: function(statementTypes) {
+        assert(
+            Array.isArray(statementTypes) || statementTypes === true,
+            'disallowCurlyBraces option requires array or true value'
+        );
+
+        if (statementTypes === true) {
+            statementTypes = defaultKeywords;
+        }
+
+        this._typeIndex = {};
+        for (var i = 0, l = statementTypes.length; i < l; i++) {
+            this._typeIndex[statementTypes[i]] = true;
+        }
+    },
+
+    getOptionName: function() {
+        return 'disallowCurlyBraces';
+    },
+
+    check: function(file, errors) {
+
+        function addError(typeString, node) {
+            var entity = node;
+
+            if (typeString === 'Else') {
+                entity = file.findPrevToken(
+                    file.getTokenByRangeStart(node.alternate.range[0]),
+                    'Keyword',
+                    'else'
+                );
+            }
+
+            errors.add(
+                typeString + ' statement with extra curly braces',
+                entity.loc.start.line,
+                entity.loc.start.column
+            );
+        }
+
+        function checkBody(type, typeString) {
+            file.iterateNodesByType(type, function(node) {
+                if (node.body && node.body.type === 'BlockStatement') {
+                    addError(typeString, node);
+                }
+            });
+        }
+
+        var typeIndex = this._typeIndex;
+
+        if (typeIndex['if'] || typeIndex['else']) {
+            file.iterateNodesByType('IfStatement', function(node) {
+                if (typeIndex.if && node.consequent && node.consequent.type === 'BlockStatement') {
+                    addError('If', node);
+                }
+                if (typeIndex['else'] && node.alternate &&
+                    node.alternate.type === 'BlockStatement' &&
+                    node.alternate.type !== 'IfStatement'
+                ) {
+                    addError('Else', node);
+                }
+            });
+        }
+
+        if (typeIndex['case'] || typeIndex['default']) {
```

Comment:
And this rule only applies to single line statements since you are required to have braces otherwise right? Yeah most people don't put curly's in their case/default anyway - Oh and that was what the original issue was about.. he wanted to remove curly's for case/default!